### PR TITLE
CIV-8967 Upload document to docstore on file upload (JS)

### PR DESCRIPTION
**Pull request by jakiurmalik**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-8967

### Change description ###
On file upload component, trigger the file validation using javascript
on validation passing, trigger async file upload to civil service
on return save CaseDocument object to the page data
Whilst waiting on the request to civil service, display an "uploading" text on the component and return when complete

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by LeonardoPalmeiro — DISMISSED** [review]

File: src/main/common/models/caseProgression/uploadDocumentsUserForm.ts [hunk 1]
```diff
@@ -1,4 +1,7 @@
-import {IsNotEmpty, ValidateNested} from 'class-validator';
+import {IsDefined, IsNotEmpty, ValidateIf, ValidateNested} from 'class-validator';
+import { IsFileSize} from 'form/validators/isFileSize';
```
Comment (hmcts-version1-ignacio) [inline src/main/common/models/caseProgression/uploadDocumentsUserForm.ts:2]: Extra space here.

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by kurt-uk — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by m-meulendijks-v1 — DISMISSED** [review]

**Review by kurt-uk — APPROVED** [review]

File: src/main/services/features/caseProgression/caseProgressionService.ts [hunk 2]
```diff
@@ -166,16 +167,20 @@ const getFormSection = <T>(data: any[], bindFunction: (request: any) => T): T[]
 const bindRequestToTypeOfDocumentSectionObj = (request: any): TypeOfDocumentSection => {
   const formObj: TypeOfDocumentSection = new TypeOfDocumentSection(request['dateDay'], request['dateMonth'], request['dateYear']);
   formObj.typeOfDocument = request['typeOfDocument'].trim();
-  //TODO we should get the file from dm-store
   formObj.fileUpload = request['fileUpload'];
+  if (request['caseDocument'] && request['caseDocument'] !== '') {
```
Comment (m-meulendijks-v1) [inline src/main/services/features/caseProgression/caseProgressionService.ts:171]: Would a constant make sense for 'caseDocument'?

File: src/main/views/features/dashboard/item-content.njk [hunk 3]
```diff
@@ -239,6 +239,20 @@
            }
         }) }}
     </div>
+    {{ govukInput({
+       id: content.data.category+'['+content.data.index+'][caseDocument]',
+       name: content.data.category+'['+content.data.index+'][caseDocument]',
+       type:'hidden',
+       label: {
+        text: 'Case Document',
```
Comment (m-meulendijks-v1) [inline src/main/views/features/dashboard/item-content.njk:247]: Does this need a t() potentially?

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by hmcts-version1-ignacio — DISMISSED** [review]

File: src/main/services/features/caseProgression/caseProgressionService.ts (hunk 2 above)
Comment (LeonardoPalmeiro) [reply src/main/services/features/caseProgression/caseProgressionService.ts:171]: yes..definitely!!

**Review by bhagyashreesharma90 — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Pull request closed without merging**